### PR TITLE
Explicate init error types

### PR DESCRIPTION
**Pull request by ShanniLi**
r @jcorbin @kriskowal 

fixed issue #571 to add explicit types for init errors.

### CODE REVIEW

**Comment by Raynos** [comment]
lgtm.

**Comment by kriskowal** [comment]
Yep, that’s the way we do TypedError. :+1:

File: node/errors.js [hunk 1]
```diff
@@ -237,10 +237,34 @@ module.exports.SocketClosedError = TypedError({
     reason: null
 });
 
-module.exports.TChannelInitProtocolError = TypedError({
-    type: 'tchannel.init.protocol-error',
-    message: 'tchannel: {reason}',
-    reason: null
+module.exports.TChannelDuplicateInitRequestError = TypedError({
```
Comment (jcorbin) [inline node/errors.js]: alphabetical order please... search indexes aren't just for silicon computers

**Comment by jcorbin** [comment]
lgtm, but please restore order to the errors module before merge

File: node/errors.js (hunk 1 above)
Comment (ShanniLi) [inline node/errors.js]: fixed.

**Pull request merged**